proto/frequenz/api/microgrid/microgrid.proto

```diff
@@ -412,15 +413,14 @@ message Component {
   // The model name of the component.
   string model_name = 5;
 
-  // The type of the component.
-  // E.g., if the `ComponentCategory` is `INVERTER`, then the inverter type
-  // (`UNSPECIFIED`, `BATTERY`, `SOLAR`, or `HYBRID`)
-  oneof type {
-    battery.Type battery = 6;
-    inverter.Type inverter = 7;
-    meter.Type meter = 8;
-    ev_charger.Type ev_charger = 9;
-    sensor.Type sensor = 10;
+  // The metadata specific to the component category.
+  oneof metadata {
+    grid.Metadata grid = 10;
```

Comment:
Shouldn't this start at 11 to avoid repeating used tags?

Maybe also add a comment here saying which tags were used in the past and should not be reused?

Reply:
> Shouldn't this start at 11 to avoid repeating used tags?

I don't think repeating used tags is a concern, because this is going to be rolled out as a breaking change. But if you really want new tags, I can update these.

Reply:
OK, it was strange that you use new tags for all the metadata except for the new `grid` one, so it looked like something not intentional.

In any way, IIRC, it is a good practice to not reuse tags in protobufs, as at the protocol level if we do if for some reason we have mismatched versions between the server and the client, and the server is sending some `grid` metadata the client might interpret it as some old `sensor` type instead, maybe making the client code blow up in a more unexpected way.

I agree with such big change it will probably blow up pretty soon and consistently but if the changes were smaller it could be a harder to find issue if versions are accidentally mixed.

In any case, I'd strongly prefer to either:
* Not reuse tags
* Keep tags continuous

Right now it is weird that we suddenly jump from 5 to 10 for no apparent reason (if you don't see this PR :).

Reply:
>so it looked like something not intentional.

I just did it to keep `10` available for `grid` in the 2nd commit :)

> at the protocol level if we do if for some reason we have mismatched versions between the server and the client, and the server is sending some grid metadata the client might interpret it as some old sensor type instead, maybe making the client code blow up in a more unexpected way.

Yes, that will happen, and I agree that we should not reuse tags (in fact it's a standard practice to not reuse tags).
But we are at a very early stage, and doing it now will blow up the tag numbers very quickly. I'd suggest waiting until we release v1.0.0 until we introduce such constraints. 

> Right now it is weird that we suddenly jump from 5 to 10 for no apparent reason (if you don't see this PR :).

True. We have left intentional gaps in places so that we can introduce new items in a non-breaking change.
But this is not required, since the numbers can be arbitrary. We can fix those in a cleanup PR.

Reply:
I'll just go ahead an renumber the tags, so that older ones are not used.

Reply:
OK, as I said before, for me consistency/information is the most important. If you want to reuse tags because it is very early in the development, then sure, but I would clarify in the commit message for example, and chose contiguous tags.

If there is a gap in the tags, either add a comment in the code saying the gap is there because there were some old attributes using those tags so they shouldn't be reused, or maybe have it as a general rule of the project that gaps can't be reused (but IMHO being explicit locally is always better and less error prone).

Reply:
Fair enough 👍